src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -88,13 +89,7 @@ public static async ValueTask<SslStream> EstablishSslConnectionAsync(SslClientAu
                     throw CancellationHelper.CreateOperationCanceledException(e, cancellationToken);
                 }
 
-                HttpRequestException ex = new HttpRequestException(SR.net_http_ssl_connection_failed, e);
-                if (request.IsExtendedConnectRequest)
-                {
-                    // Extended connect request is negotiating strictly for ALPN = "h2" because HttpClient is unaware of a possible downgrade.
-                    // At this point, SSL connection for HTTP / 2 failed, and the exception should indicate the reason for the external client / user.
-                    ex.Data["HTTP2_ENABLED"] = false;
```

Comment:
The existence of these data entries is unfortunately a part of the contract at this point since we shipped it in .NET 7. I don't think we can remove them (at least not in Net 8).

[YARP relies on that](https://github.com/microsoft/reverse-proxy/blob/f74196b1f643e422f5f95f6aab6e39dde66eabfc/src/ReverseProxy/Forwarder/HttpForwarder.cs#L166-L180) in its HTTP/2 WebSocket implementation. Removing them would break the automatic version downgrade if users started running on .NET 8.

(same for the other 2 places that modified these)